src/htm/types/Sdr.hpp

```diff
@@ -471,10 +471,16 @@ class SparseDistributedRepresentation : public Serializable
      * @param rng The random number generator to draw from.  If not given, this
      * makes one using the magic seed 0.
      */
-    void addNoise(Real fractionNoise);
+    void addNoise(Real fractionNoise); //TODO the name is confusing, rename to shuffle ?
```

Comment:
is it OK to rename this to shuffle() and have addNoise for the new fn? @ctrl-z-9000-times

Reply:
I looked at your new `addNoise` function and I think it will have issues with keeping the sparsity at a reasonable level.  I think that the sparsity of an SDR after this method is called on it will always tend towards 50%.

Reply:
that would be indeed wrong. What I intended: 
- have SDR of current input
  - flip 0.01% bits
- have a new SDR
  - flip 0.01%bits

So the sparsity would remain the same (actially grow, because we have much more off bits, so flipping on would be more probable). But it should remain the x% (2%) + 0.001%